src/Listener/UpdateMentionsMetadataWhenVisible.php

```diff
@@ -43,14 +43,18 @@ public function handle($event)
     {
         $content = $event->post->parsedContent;
 
+        $canMention = function (string $type) use ($event) {
+            return $event->actor->can("mention.$type", $event->post->discussion);
+        };
+
         $this->syncUserMentions(
             $event->post,
-            Utils::getAttributeValues($content, 'USERMENTION', 'id')
+            $canMention('users') ? Utils::getAttributeValues($content, 'USERMENTION', 'id') : []
```

Comment:
Instead of the `?:` statement here, can we surround the whole `sync` call with an `if`?

Reply:
I've changed the code to make it easier to read.